jobs/haproxy/spec

```diff
@@ -357,12 +347,29 @@ properties:
           The client certificate received by HAProxy in the mTLS handshake will not be forwarded.
```

Comment:
We also delete the `X-SSL-Cert-xxx` headers on non-mTLS connections here. We should mention that.

Reply:
Fixed